gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/util/CodeableConceptUtil.java

```diff
@@ -0,0 +1,46 @@
+package uk.nhs.adaptors.pss.translator.util;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.springframework.util.CollectionUtils;
+import org.hl7.v3.CD;
+import java.util.List;
+import java.util.Objects;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class CodeableConceptUtil {
+
+    public static Boolean compareCodeableConcepts(CD c1, CD c2) {
+
+        if (c1 == null || c2 == null) {
+            return c1 == c2;
+        }
+
+        return Objects.equals(c1.getCode(), c2.getCode())
+                && Objects.equals(c1.getCodeSystem(), c2.getCodeSystem())
+                && Objects.equals(c1.getDisplayName(), c2.getDisplayName())
+                && Objects.equals(c1.getOriginalText(), c2.getOriginalText())
+                && compareCodeSubElements(c1.getQualifier(), c2.getQualifier())
+                && compareCodeSubElements(c1.getTranslation(), c2.getTranslation());
+    }
+
+    private static Boolean compareCodeSubElements(List<? extends CD> c1, List<? extends CD> c2) {
+
+        if (CollectionUtils.isEmpty(c1) || CollectionUtils.isEmpty(c2) || c1.size() != c2.size()) {
+            return CollectionUtils.isEmpty(c1) && CollectionUtils.isEmpty(c2);
+        }
```

Comment:
I feel like our tests will continue to pass, and the code simplier, if we just replaced this with:

```suggestion
        if (c1.size() != c2.size()) {
            return false;
        }
```

Reply:
I've done the change that was suggested